### PR TITLE
[MHV-45995, MHV-45991] Alt text on Message Icons, Sensory characteristics updated

### PR DESCRIPTION
**Pull request by vsaleem**
## Summary
- Resolved alt text for sm icons - (Paperclip `Has Attachment` and Fa-circle `Unread message`)
- Updated sensory characteristics in message thread view - Removed `<H2>` role on `<main/>`

## Related issue(s)
JIRA:
- (https://jira.devops.va.gov/browse/MHV-45991)
- (https://jira.devops.va.gov/browse/MHV-45995)
- (https://jira.devops.va.gov/browse/MHV-45987)
- (https://jira.devops.va.gov/browse/MHV-45986)
- (https://jira.devops.va.gov/browse/MHV-45990)

GITHUB ISSUES:
- (https://github.com/department-of-veterans-affairs/va.gov-team/issues/59462)
- (https://github.com/department-of-veterans-affairs/va.gov-team/issues/59461)
-  (https://github.com/department-of-veterans-affairs/va.gov-team/issues/59457)

## Testing done

- SQA Testing
- Cypress Testing

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*My Health - Secure Messaging*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/secure-messaging/components/ThreadList/ThreadListItem.jsx [hunk 1]
```diff
@@ -81,11 +81,14 @@ const ThreadListItem = props => {
...
-            <i
-              aria-hidden="true"
-              className="unread-icon vads-u-margin-right--1 vads-u-color--primary-darker fas fa-solid fa-circle"
-              data-testid="thread-list-unread-icon"
-            />
+            <span>
+              <i
+                className="unread-icon vads-u-margin-right--1 vads-u-color--primary-darker fas fa-solid fa-circle"
+                data-testid="thread-list-unread-icon"
+                alt="Unread message"
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/ThreadList/ThreadListItem.jsx:88]: Just flagging if `role="img"` needs to be added here as well as in instance above 🤷🏻‍♂️

File: src/applications/mhv/secure-messaging/components/ThreadList/ThreadListItem.jsx [hunk 2]
```diff
@@ -121,10 +132,27 @@ const ThreadListItem = props => {
...
         </Link>
+
         <p className="received-date vads-u-margin-y--0p5">
-          {hasAttachment && <i className={attachmentClasses} aria-hidden />}
+          {hasAttachment && (
+            <span
+              role="img"
+              aria-labelledby={`message-link-has-attachment-${messageId}`}
+            >
+              <i className={attachmentClasses} alt="Has attachment" />
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/ThreadList/ThreadListItem.jsx:153]: Just flagging if role="img" needs to be added here as well as in instance above 🤷🏻‍♂️

File: src/applications/mhv/secure-messaging/components/AttachmentsList.jsx [hunk 3]
```diff
@@ -92,9 +92,17 @@ const AttachmentsList = props => {
...
+                      aria-labelledby="has-attachment"
+                      className="fas fa-paperclip"
+                      aria-hidden="true"
+                      alt="Has attachment"
+                    />
+                    <span id="has-attachment" ref={attachmentReference}>
+                      {file.name}
+                    </span>
+                    ({getSize(file.size || file.attachmentSize)})
+                    <span className="sr-only">Has attachment</span>
```
[the remark this answers is not in the document]
Comment (oleksii-morgun-va) [reply src/applications/mhv/secure-messaging/components/AttachmentsList.jsx:105]: Maybe "Attachment icon" would be more applicable in this case? Could be a good question to ask on that same github issue ticket or reach out directly to a person who raised this

**Review by alex-morgun-1 — APPROVED** [review]
looks great!

**Review by mdewey — APPROVED** [review]
LGTM. This is a step in the right direction!

**Pull request merged**